src/app/api/core/types/message.ts

```diff
@@ -0,0 +1,24 @@
+import { z } from 'zod'
+import { SyncStatus } from '@prisma/client'
+
+export const ChannelSchema = z.object({
+  id: z.string(),
+  createdAt: z.string().datetime(),
+  lastMessageDate: z.string().datetime().nullable(),
+  memberIds: z.array(z.string().uuid()),
+  membershipEntityId: z.string().uuid(),
+  membershipType: z.enum(['individual', 'group', 'company']),
```

Comment:
we may need to use it other places as well. so a separate enum would have been better